plugins/languages/lean.nix

```diff
@@ -0,0 +1,316 @@
+        '';
+
+        extra = helpers.defaultNullOpts.mkNullable (with types; attrsOf str) "{}" ''
+          Additional abbreviations.
+
+          Example:
+          ```
+            {
+              # Add a \wknight abbreviation to insert ♘
+              #
+              # Note that the backslash is implied, and that you of
+              # course may also use a snippet engine directly to do
+              # this if so desired.
+              wknight = "♘";
+            }
+        '';
+
+        leader = helpers.defaultNullOpts.mkStr "\\" ''
+          Change if you don't like the backslash.
+          (comma is a popular choice on French keyboards)
+        '';
+      };
+
+      mappings = helpers.defaultNullOpts.mkBool false ''
+        Whether to enable suggested mappings.
+      '';
+
+      infoview = {
+        autoopen = helpers.defaultNullOpts.mkBool true ''
+          Automatically open an infoview on entering a Lean buffer?
+          Should be a function that will be called anytime a new Lean file is opened.
+          Return true to open an infoview, otherwise false.
+          Setting this to `true` is the same as `function() return true end`, i.e. autoopen for any
+          Lean file, or setting it to `false` is the same as `function() return false end`,
+          i.e. never autoopen.
+        '';
+
+        autopause = helpers.defaultNullOpts.mkBool false "";
+
+        width = helpers.defaultNullOpts.mkPositiveInt 50 ''
+          Set infoview windows' starting width.
+          Windows are opened horizontally or vertically depending on spacing.
+        '';
+
+        height = helpers.defaultNullOpts.mkPositiveInt 20 ''
+          Set infoview windows' starting height.
+          Windows are opened horizontally or vertically depending on spacing.
+        '';
+
+        horizontalPosition = helpers.defaultNullOpts.mkEnum ["top" "bottom"] "bottom" ''
+          Put the infoview on the top or bottom when horizontal.
+        '';
+
+        separateTab = helpers.defaultNullOpts.mkBool false ''
+          Always open the infoview window in a separate tabpage.
+          Might be useful if you are using a screen reader and don't want too many dynamic updates
+          in the terminal at the same time.
+          Note that `height` and `width` will be ignored in this case.
+        '';
+
+        indicators = helpers.defaultNullOpts.mkEnum ["always" "never" "auto"] "auto" ''
+          Show indicators for pin locations when entering an infoview window.
+          `"auto"` = only when there are multiple pins.
+        '';
+
+        lean3 = {
+          showFilter = helpers.defaultNullOpts.mkBool true "";
+
+          mouseEvents = helpers.defaultNullOpts.mkBool false ''
+            Setting this to `true` will simulate mouse events in the Lean 3 infoview, this is buggy
+            at the moment so you can use the I/i keybindings to manually trigger these.
+          '';
+        };
```

Comment:
shouldn this be in the global scope  and not in `infoview` ? (as from the README.rst):

```lua
  -- Redirect Lean's stderr messages somehwere (to a buffer by default)
  stderr = {
    enable = true,
    -- height of the window
    height = 5,
    -- a callback which will be called with (multi-line) stderr output
    -- e.g., use:
    --   on_lines = function(lines) vim.notify(lines) end
    -- if you want to redirect stderr to `vim.notify`.
    -- The default implementation will redirect to a dedicated stderr
    -- window.
    on_lines = nil,
  },

  -- Legacy Lean 3 support (on_attach is as above, your LSP handler)
  lsp3 = { on_attach = on_attach },

  -- mouse_events = true will simulate mouse events in the Lean 3 infoview, this is buggy at the moment
  -- so you can use the I/i keybindings to manually trigger these
  lean3 = { mouse_events = false },
}
```

Reply:
This is indeed what the documentation suggests, but after looking at the code, this is clearly under `infoview`.
This isn't the only error I spotted in this documentation section.

Reply:
Indeed.... I tried to search for a compatibility layer but there does not seem to be any